design/one-pager-definition-of-done.md

```diff
@@ -0,0 +1,46 @@
+# Definition of done
+
+* Owner: Daniel Suskin (@suskin)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Introduction
+
+*Why does this document exist?*
+
+The goal of this document is to answer the questions:
+
+* As a developer, when should I consider a chunk of work to be complete?
+* As a reviewer, what should I be looking for to verify that a chunk of work is
+  complete?
+
+## Definition of work
+
+Work means a contribution to the project, typically code.
+
+## Definition of done
+
+The following criteria **MUST** be met for work to be considered done:
+
+* Code is reviewed and approved
+* Code passes linting (use `make reviewable` to lint)
+* All error conditions are reported into the log or as an event, as
+  appropriate
```

Comment:
This feels overly prescriptive to me. I would prefer this sentence without the "all", and the details i.e. "Error conditions are reported where appropriate". Given how errors work in Go, I think it would be easy to interpret this as "every time you find `err != nil` you must log it", which is not a position I agree with.

https://www.youtube.com/watch?v=PAAkCSZUG1c&t=16m13s
https://blog.golang.org/errors-are-values

I'd prefer to keep this vague for now, then come back and update it when we ratify your proposal around error handling.

Reply:
The previous section ends ", typically code."

Bullets 1,2, and 4 assert that this applies to "code" change, but bullet 3 does not -- it is both too specific (exact use of errors) and not specific enough (only for code?).

We may need different "definitions of done" sections for: code, design, documentation, examples, artwork, build artifacts (and Stacks). And what of meta-work like issue and epic creation and preening?

We could refer to existing guides when possible.

On the other hand, if we can keep this generic, we may not need to spell out DoD for each type of work.